File: src/WebApp/src/app/inventory/interfaces/note.ts
import {ItemNote} from './item-note';

export class Note {
  constructor(
    public contactId: number = 0, // Id de Contacto.
    public warehouseId: number = 0, // Id del Almacén.
    public noteType: string = '', // Tipo de Nota (Ingreso|Salida).
    public motivo: number | string = 0, // Motivo de Inventario.
    public startDate: any = null, // Fecha de registro.
    public itemNotes: Array<ItemNote> = new Array<ItemNote>()) {
  }
}

File: src/Nebula/Data/ViewModels/Note.cs
namespace Nebula.Data.ViewModels;

/// <summary>
/// Nota de Inventario.
/// </summary>
public class Note
{
    /// <summary>
    /// Id de Contacto.
    /// </summary>
    public int ContactId { get; set; }

    /// <summary>
    /// Id del Almacén.
    /// </summary>
    public int WarehouseId { get; set; }

    /// <summary>
    /// Tipo de Nota (Ingreso|Salida).
    /// </summary>
    public string NoteType { get; set; }

    /// <summary>
    /// Motivo de Inventario.
    /// </summary>
    public int Motivo { get; set; }

    /// <summary>
    /// Fecha de registro.
    /// </summary>
    public DateTime StartDate { get; set; }

    /// <summary>
    /// Observación de la Nota.
    /// </summary>
    public string Remark { get; set; }

    /// <summary>
    /// Detalle de la Nota.
    /// </summary>
    public List<ItemNote> ItemNotes { get; set; }
}
